### PR TITLE
feat: [GALI-2097] DataSetAuthService

### PR DESCRIPTION
**Pull request by aws-tyler**
Issue #, if available:
GALI-2097

Description of changes:
Added a mock implementation of the `DataSetsAuthorizationPlugin` and integrated it into the DataSetService

### CODE REVIEW

File: solutions/swb-app/src/dataSets/dataSetPermissionParser.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { z } from 'zod';
+
+// eslint-disable-next-line @rushstack/typedef-var
+export const DataSetPermissionParser = z
+  .object({
+    subject: z.string(),
+    accessLevel: z.union([z.literal('read-only'), z.literal('read-write')])
```
Comment (FernandoAranda) [inline solutions/swb-app/src/dataSets/dataSetPermissionParser.ts:12]: does `z.enum` not work for this?

**Review by FernandoAranda — APPROVED** [review]

File: solutions/swb-app/src/dataSets/dataSetPermissionParser.ts (hunk 1 above)
Comment (aws-tyler) [reply solutions/swb-app/src/dataSets/dataSetPermissionParser.ts:12]: The accessLevel type is actualy the union of two string literals, not an enum, so I tried to match that exactly without having to add another parser.

**Review by FernandoAranda — APPROVED** [review]

File: solutions/swb-app/src/dataSets/dataSetPermissionParser.ts (hunk 1 above)
Comment (FernandoAranda) [reply solutions/swb-app/src/dataSets/dataSetPermissionParser.ts:12]: Interesting so this translates into `'read-only' | 'read-write'`

Comment (bhudnell) [reply solutions/swb-app/src/dataSets/dataSetPermissionParser.ts:12]: `z.enum` seems to be the Zod way of creating a union of string literals. https://github.com/colinhacks/zod#zod-enums

Comment (aws-tyler) [reply solutions/swb-app/src/dataSets/dataSetPermissionParser.ts:12]: :-( why can't we just have nice enums like everyone else

**Pull request merged**